README.md

```diff
@@ -0,0 +1,116 @@
+# Devise: Setup and customization
+
+Set up a Rails server with [Devise] to let users log in with a username and password.
+
+## Objectives
+
+We're going to use Devise and Rails to create an app with a complete signin flow. We will basically be able to do this using only generators.
+
+## Instructions: Part 1, setup
+
+Add Devise to your Gemfile:
+
+    gem 'devise'
+
+Now run the installer:
+
+    rails generate devise:install
+
+This creates a massive initializer in `config/initializers/devise.rb`. This is probably the single best source of documentation for Devise that I've seen. You may wish to look through it.
+
+You'll notice that the installer prints a big notice of several things you should do. In particular, we should have a route route.
+
+Create a `WelcomeController` with a `home` action and a view that just prints `current_user.email`.
+
+Now generate your `User` model with:
+
+    rails g devise User
+
+Run `rake routes`. You should see that Devise has added a bunch of them. Run `rails s` and take a look at one, maybe `/users/sign_in`.
+
```

Comment:
There is a migration step missing here. The error would be obvious. Not sure how explicit we need to be. @jmburges?